### ISSUE TITLE
Installer slows system until frozen

### ISSUE THREAD

**Issue by AlexTes** [Antergos/Cnchi#270]
I thought I'd found my problem in the issues at least three times, but you guys fix problems pretty quick. I was hoping to use Antergos because I didn't feel like going through installing Arch again now that I need to reinstall, but this is taking way more time anyway, go figure. You have me hooked on your default for gnome as a second to my bspwm however so it's all good.

As described the install rolls over fine until ~~it tries to download the linux-api-headers.~~ a random point early during downloading. Rerunning with `sudo -E cnchi -dv` showed everything as normal, downloads rolling along. As the installer freezes the system is there anyway to make sure nothing more was written to log?

The install was freezing half the time a week back when I was installing too, ~~I'm starting to suspect the USB.~~

Background information:
Running from a 16gb 3.0 USB
Booting in UEFI mode though the bios has CSM enabled
I've copied a fresh downloaded image twice. 
Tried the minimal ISO (this one can't ask for wifi passwords btw)
Tried different USB 3.0 ports and a 2.0 port.
Tried over LAN and WiFi.
EDIT: Tried with two separate USB sticks.

I have one SSD one 1TB harddisk, both gpt tabled.
With gparted I've formatted the SSD to have one 312MB FAT32 partition. One main ext4 partition for the system files and at the end 8gb formatted to linux-swap.
The second drive's one and only partition gets mounted as my home and *not* formatted. With the rest I've tried formatting and not formatting in every possible configuration.

**Comment by AlexTes** [comment]
Just like #240 my laptop starts increasing fan speed as if in some sort loop problem.
This time the last thing in the log were circular dependency warnings. 
Unlikely to be related though.

**Comment by AlexTes** [comment]
Also tried letting cnchi do the partition creation. Got all the way to the installer configuring the system!! Last two lines visible read 'creating user...' and '...fingerprint ...' before dying again.

**Comment by lots0logs** [comment]
First thing I want to mention is that you should reboot your system after a failed attempt to install. Chances are very slim that it will complete on a second attempt without fresh reboot,

Could you post the log? You can find it at `/tmp/cnchi.log` 

Thanks!

**Comment by AlexTes** [comment]
I'm confused. My system freezes. Entirely. It slowly grinds to a complete unresponsive halt. In other words, the only thing I can do after I have this problem is a hard shutdown. Hence my question on how to recover the log. I'm not sure what you mean exactly by reboot. If a hard shutdown and straight boot from the flash drive again qualify as reboot, I have been doing that, its the only thing I can do.

Also for the log, my question thus remains, after hard shut down and reboot the log only shows the current cnchi's output.

**Comment by lots0logs** [comment]
Can you switch to another VT after it freezes? `Ctrl + Alt + F2`?

**Comment by AlexTes** [comment]
No, unfortunately not. I don't think it freezes as in really hanging, I thought that at first, but now I often catch it getting slower, the mouse position updating about once every two seconds, until it doesn't at all. I've waited a minute after trying to switch before, but nothing I'm afraid.

**Comment by AlexTes** [comment]
Tried the VT switch before install to make sure everything works. It does, but it freezes after about a minute. USB flikkers io, fans spin up, system dies. tty2 prints a message though.
`INFO: rcu_preempt detected stalls on CPUs/tasks: { 0} (detected by 2, t=18002 jiffies, g=2903, c=2902, q=629)`
then a hundred seconds later reading from the timestamp:
`INFO: task loop0:175 blocked for more than 120 seconds.
Tainted: G   W 0   3.17.6-1-ARCH #1
"echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.`
After this it prints the same with the exception that "loop0:175" has been replaced with other tasks.

Is it the USB after all? They both have seen a lot of flashing and have housed windows installer stuff. Going to try and do some more testing on my USB's.

**Comment by lots0logs** [comment]
It seems like it has something to do with the USB. Try a different USB and use a different port (one on a separate bus if possible)

**Comment by AlexTes** [comment]
I am trying two (same model) different USB's. I've already checked the main one I'm using for bad sectors, wiped it completely and it really seems fine. Tried the Arch image because I wondered whether nouveau problems systemd diplays during boot and shutdown had anything to do with it, Arch seems stable but I haven't tried a full Arch install yet, just booting it.

I did just try the antergos cli installer through tty2, it failed to make ext4 on my ssd. Gave the same 'Tainted' errors. Has that been the problem the whole time? it downloads packages parallel to making the filesystem and at a point just runs into problems on my ssd? Either during the creation of the filesystem or at a later point when there IO going on with that disk.

**Comment by AlexTes** [comment]
Making the filesystem with Arch (mkfs.ext4) on the ssd went fine. It's installing the system now, everything fine and dandy. I don't think its the USB. Is there anything I can do to make sure its my system or antergos. I'd like to help. And I'd like your beautifully configured gnome environment as one of my sessions.

**Comment by AlexTes** [comment]
If it's any help, these are the systemd messages displayed when hanging on shutdown.
[picture of output](http://i.imgur.com/08499rT.jpg)

**Comment by lots0logs** [comment]
What are your system specs? It could be a kernel bug. Also, are you using advanced mode to create your own partitions? If so, what's the layout (including filesystem types)?

**Comment by AlexTes** [comment]
Body: Clevo w35xstq_370st
Processor (CPU): Intel® CoreTMi7 Quad Core Mobile Processor i7-4700MQ (2.40GHz) 6MB
Memory (RAM): 8GB SAMSUNG 1600MHz SODIMM DDR3 MEMORY (1 x 8GB)
Graphics Card: NVIDIA® GeForce® GTX 765M - 2.0GB DDR5 Video RAM - DirectX® 11
1st Hard Disk: 120GB KINGSTON V300 SSD, SATA 6 Gb (450MB/R, 450MB/W)
2nd Hard Disk: 1TB SERIAL ATA II 2.5" HARD DRIVE WITH 8MB CACHE (5,400rpm)

It's an optimus laptop. Arch just gave me the nouveau error as well btw.

I've tried both using gparted to make the structure as I wanted and the installer. I've tried labelling everything, nothing, and formatting everything nothing (except system). Assume I'm formatting everything except the data partition. That's what I tried most.
Layout: [device] [size] [type] [mountpoint]
/dev/sda1 512MB fat32 /boot
/dev/sda2 111GB ext4 /
/dev/sda3 8GB swap
/dev/sdc1 1TB /home

**Comment by lots0logs** [comment]
Hmm..The nouveau driver has been buggy the last few kernel releases. I'm using the nvidia drivers because nouveau wouldnt act right. There's currently no way to switch drivers on the ISO because it would require a reboot and there's no persistent storage for the live system. I'll try to add the nvidia drivers on the iso but I don't know exactly when I will have time to do it. 

If you have already installed Arch just go ahead and switch to the nvidia drivers and see if gnome will start. If it does, I can tell you how to run our setup script for the default gnome configuration :smile:

**Comment by AlexTes** [comment]
Oh gnome starts fine! I don't know how, and why I see messages of nouveau complaining loginctl crashing and then getting a fully functioning graphical system (until it grinds to a halt minutes later). But as Arch also gives me them I don't think those are the problem.

I'm really curious to find what it is thats going wrong, its clearly something to do with my system or others would've complained about 6.50 by now I'm guessing. I'm almost done getting Arch up and I would absolutely LOVE to add your gnome configuration through a setup script post installation. But when I have time again I think I'll just backup Arch and start fiddling with Antergos again. I'm curious whats failing, writing a script to upload the log every few seconds is a good reason to finally learn some more shell scripting :D Maybe you see something in the log I don't.

**Comment by AlexTes** [comment]
Got Arch working now. Would love to hear how to get gnome going with the Antergos config!

Feel free to close this issue, it might be a while until I get to backing up arch and playing with Antergos again.

**Comment by lots0logs** [comment]
Okay, here's what you need...

https://github.com/Antergos/Cnchi/blob/master/scripts/postinstall.sh
https://github.com/Antergos/Cnchi/blob/master/scripts/set-settings

Before you run the script you will need to ensure you have all the packages installed for our gnome setup (the packages listed under `graphical` and `gnome`) in this file:

https://github.com/Antergos/Cnchi/blob/master/data/packages.xml

In postinstall.sh you need to change this line so that it copies the set-settings file from whereever you downloaded to: `cp /usr/share/cnchi/scripts/set-settings ${DESTDIR}/usr/bin/set-settings`

Then run the script as root with the following arguments:
1. your user name
2. destination dir - for you it will be `""`
3. desktop (gnome)
4. keyboard layout
5. keyboard variant

So putting it all together it would look something like this:

```
sudo ./postinstall.sh dustin "" gnome en US
```

Good luck :smile:

**Comment by prescott66** [comment]
Maybe is arch issue...i get sometimes freezes while using installed antergos. Hdd led glows constantly, mouse cursor is slowed down...

**Comment by AlexTes** [comment]
Maybe, I'm quite sure I fixed it though! I haven't got the time to nail it down exactly, but changing some BIOS settings to do with USB got me to install without freeze, succeeded twice (wanted to make sure my USB stick wasn't the problem, was testing with roommate's initially) as long as I used the USB port I normally don't test on.

**Comment by prescott66** [comment]
what for a settings you have changed in bios?

**Comment by AlexTes** [comment]
My laptop has two hubs, the one on the left side is USB 2.0, I used that one, Left legacy USB suport on, 3.0 support(XHCI controller support) off, disabled XHCI handoff EHCI handoff and have USB Mass Storage Driver Support enabled to. That last one is needed in any case if you want to install from USB.

I'm not sure which of these settings matter and which don't again I haven't tested, it might just be the two times I tried the left 2.0 port I also changed some BIOS settings that crashed Antergos on start, I've tried too many different settings too many times to remember what the exact config was the couple times I tried the 2.0 port. So what I'm trying to say is, I'm quite certain its one of these settings, quite possibly just using the 2.0 port fixed it.

How all of this gives trouble with Antergos and not with Arch I'm unsure. Let me know if I can be of more help @lots0logs, and feel free to close the issue if you don't want to take action on this still very unclear problem.
